SNT/NativeMemoryManager.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System.Buffers;
+using System.Runtime.InteropServices;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System;
+
+namespace Torch.SNT
```

Comment:
Should we be using non-descriptive names like 'SNT' in namespaces? It does not aid in discovery.

Reply:
I agree. Any suggestion? I can definitely unroll SNT as SystemNumericTensor but it becomes really long.